### PR TITLE
CORE-2020

### PR DESCRIPTION
**Pull request by TheChronicMonster**
Added documentation for RPC access to Views, APIs, and Events

Restructured presentation of information in RPC document and added new headings for a stronger structure.

### CODE REVIEW

File: docs/src/rpc/index.md [hunk 1]
```diff
@@ -28,31 +30,71 @@ All `/stdlib` methods are synchronous value RPC methods that accept and produce
...
+
+For Example:
+
+```
+rpc('/ctc/views/', view_creator)      # Access view 'Creator'
+rpc('/ctc/apis/', bidder)             # Call 'Bidder' API
+rpc('/ctc/events/', event_bid)        # Access 'Bid' event object
+```
+
+Below, in line 45, `getRead` accesses a `{!rsh} View` named `Reader` from the attached contract stored in `const c`.
```
Comment (mattaudesse) [inline docs/src/rpc/index.md:65]:
```suggestion
Below, in line 45, `getRead` accesses a `{!rsh} View` named `Reader` from the attached contract handle stored in `const c`. 
```

File: docs/src/rpc/index.md [hunk 2]
```diff
@@ -28,31 +30,71 @@ All `/stdlib` methods are synchronous value RPC methods that accept and produce
...
+
+``` js
+load: /examples/rpc-api-full/index.mjs
+md5: c670a626bd1b081f278ffbeb128794c9
+range: 59-63
+```
+
+By providing access to `{!rsh} View`s, `{!rsh} API`s, and `{!rsh} Events`,  RPC `/ctc` methods allow developers to build complete frontend Reach applications in the language of their choice. 
+
+### {#ref-part-methods} PART Methods
```
Comment (mattaudesse) [inline docs/src/rpc/index.md:83]: Another label.

File: docs/src/rpc/index.md [hunk 3]
```diff
@@ -28,31 +30,71 @@ All `/stdlib` methods are synchronous value RPC methods that accept and produce
...
 
 `/ctc/p/$PARTICIPANT` and `/ctc/participants/$PARTICIPANT` are interactive RPC method alternatives to the `/backend/$PARTICIPANT` method described below.
 They accept the same arguments and behave identically.
 
 + `/forget/token` accepts a token RPC handle and deletes it from the Reach RPC Server's memory.
 + `/backend/$PARTICIPANT` where `$PARTICIPANT` is a participant of the backend compiled by the [JavaScript backend](##ref-backends-js).
 
+### {#ref-backend-methods} Backend Methods
```
Comment (mattaudesse) [inline docs/src/rpc/index.md:91]: Another label.

**Review by mattaudesse — CHANGES_REQUESTED** [review]

File: docs/src/rpc/index.md [hunk 4]
```diff
@@ -28,31 +30,71 @@ All `/stdlib` methods are synchronous value RPC methods that accept and produce
 Those methods instead accept and produce account RPC handles, which are random strings that represent the corresponding account representations.
 For example, `/stdlib/newTestAccount` does not return an account like `{!js} newTestAccount`, but instead returns an account RPC handle.
 
-+ `/forget/acc` accepts an account RPC handle and deletes it from the Reach RPC Server's memory.
-+ `/acc/$METHOD` where `$METHOD` is a method of an account representation of the [JavaScript standard library](##ref-frontends-js).
+### {#ref-acc-methods} ACC Methods
```
[the remark this answers is not in the document]
Comment (jeapostrophe) [reply docs/src/rpc/index.md:33]: And don't say "ACC"... say "account"

File: docs/src/rpc/index.md [hunk 5]
```diff
@@ -29,30 +31,79 @@ Those methods instead accept and produce account RPC handles, which are random s
...
+### {#ref-rpc-method-acc} Account Methods
 
 All `/acc` methods are synchronous value RPC methods that accept and produce the same arguments and return values as the corresponding function, encoded as JSON objects, except they accept an additional first argument, which is the account RPC handle returned by a prior RPC method invocation; and, a method that accepts a backend (like `/acc/attach` (i.e. `{!js} acc.attach`) or `/acc/deploy` (i.e. `{!js} acc.deploy`) does not accept a backend argument, but has it implicitly provided by the Reach RPC Server.
 
++ `/acc/$METHOD` where `$METHOD` is a method of an account representation of the [JavaScript standard library](##ref-frontends-js).
+
 Furthermore, those that produce contract representations instead produce contract RPC handles.
 For example, `/acc/deploy` does not return a contract representation like `{!js} acc.deploy`, but instead returns a contract RPC handle.
 
 + `/forget/ctc` accepts a contract RPC handle and deletes it from the Reach RPC Server's memory.
```
Comment (mattaudesse) [inline docs/src/rpc/index.md:44]: I don't think this belongs here now.

File: docs/src/rpc/index.md [hunk 6]
```diff
@@ -29,30 +31,79 @@ Those methods instead accept and produce account RPC handles, which are random s
...
+### {#ref-rpc-method-ctc} Contract Methods
 
 Most `/ctc` methods are synchronous value RPC methods that accept and produce the same arguments and return values as the corresponding function, encoded as JSON objects, except they accept an additional first argument, which is the contract RPC handle returned by a prior RPC method invocation.
 
++ `/ctc/$METHOD` where `$METHOD` is a method of a contract representation of the [JavaScript standard library](##ref-frontends-js).
+
+`/ctc` methods can access `{!js} ctc.views`, `{!js} ctc.apis`, and `{!js} ctc.events`. 
+Similar to standard library functions, the full name or first letter can be used when calling the following methods.
+
++ `/ctc/v` or `/ctc/views` provides access to all `{!rsh} View`s.
```
Comment (mattaudesse) [inline docs/src/rpc/index.md:55]:
```suggestion
+ `/ctc/v` and `/ctc/views` provide access to `{!rsh} View`s.
```

File: docs/src/rpc/index.md [hunk 7]
```diff
@@ -29,30 +31,79 @@ Those methods instead accept and produce account RPC handles, which are random s
...
++ `/ctc/a` or `/ctc/apis` makes `{!rsh} API` functions available to the frontend.
```
Comment (mattaudesse) [inline docs/src/rpc/index.md:56]:
```suggestion
+ `/ctc/a` and `/ctc/apis` make `{!rsh} API` functions available to the frontend.
```

File: docs/src/rpc/index.md [hunk 8]
```diff
@@ -29,30 +31,79 @@ Those methods instead accept and produce account RPC handles, which are random s
...
++ `/ctc/e` or `/ctc/events` provides access to `{!rsh} Events`.
```
Comment (mattaudesse) [inline docs/src/rpc/index.md:57]:
```suggestion
+ `/ctc/e` and `/ctc/events` provide access to `{!rsh} Events`.
```

File: docs/src/rpc/index.md [hunk 9]
```diff
@@ -29,30 +31,79 @@ Those methods instead accept and produce account RPC handles, which are random s
...
+
+Any `{!rsh} View`, `{!rsh} API` or `{!rsh} Events` may be accessed with the following syntax:
+
+```
+RPC /ctc/views/$VIEW_NAME            # Top-level data of a view, $VIEW_NAME
```
Comment (mattaudesse) [inline docs/src/rpc/index.md:62]: I don't really understand the comment here.

Comment (TheChronicMonster) [reply docs/src/rpc/index.md]: I've simplified comments and added examples for APIs and Events

File: docs/src/rpc/index.md (hunk 5 above)
Comment (TheChronicMonster) [reply docs/src/rpc/index.md]: Moved to top of Account Methods section

**Comment by mattaudesse** [comment]
@jeapostrophe do you have any thoughts before JP merges this?

**Review by jeapostrophe — APPROVED** [review]

**Pull request merged**